### PR TITLE
Generalize `FtpUploader`

### PR DESCRIPTION
**Pull request by rkondratowicz**
So that downloading files can be added easier (coming soon)

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/slc/services/steps/sftpupload/FtpClient.java [hunk 1]
```diff
@@ -58,8 +60,23 @@ public FtpUploader(
...
+                String path = String.join("/", this.targetFolder, pdfDoc.filename);
+                fileTransfer.upload(pdfDoc, path);
+                insights.trackFtpUpload(Duration.between(start, Instant.now()), true);
+
+                return null;
+
+            } catch (IOException exc) {
+                insights.trackException(exc);
+                throw new FtpStepException("Unable to upload PDF.", exc);
+            }
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/slc/services/steps/sftpupload/FtpClient.java]: lost the failed insight metric

File: src/main/java/uk/gov/hmcts/reform/slc/services/steps/sftpupload/FtpClient.java [hunk 2]
```diff
@@ -58,8 +60,23 @@ public FtpUploader(
     public void upload(PdfDoc pdfDoc) {
         Instant start = Instant.now();
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/slc/services/steps/sftpupload/FtpClient.java:61]: i'd still make it null and assign in try block - beginning of transfer

**Review by doncem — DISMISSED** [review]
:clap: :rocket:

**Review by doncem — APPROVED** [review]
⛵️

File: src/main/java/uk/gov/hmcts/reform/slc/services/steps/sftpupload/FtpClient.java (hunk 1 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/slc/services/steps/sftpupload/FtpClient.java]: done

**Pull request merged**